nifi-docs/src/main/asciidoc/administration-guide.adoc

```diff
@@ -281,6 +281,272 @@ After running the client you will have the CA’s certificate, a keystore, a tru
 
 For a client certificate that can be easily imported into the browser, specify: `-T PKCS12`
 
+==== Using An Existing Intermediate Certificate Authority (CA)
+
+In some enterprise scenarios, a security/IT team may provide a signing certificate that has already been signed by the organization's certificate authority (CA). This *intermediate CA* can be used to sign the *node* (sometimes referred to as *leaf*) certificates that will be installed on each NiFi node. In order to inject the existing signing certificate into the toolkit process, follow these steps:
```

Comment:
Should we add a note about the addition of the root CA in the truststore? Or am I wrong when saying that all the elements of the trust chain must be added in the truststore?

Reply:
I beleive it depends on that the root is already added to the Java default truststore or not.

Reply:
To clarify:

* The *intermediate CA* (let's call it `intermediateca.company.com`) is signed by an (offline) *root CA* (`rootca.company.com`). 
* The *intermediate CA* is exported to `nifi-cert.pem` (public cert) and `nifi-key.key` private key. If the company doesn't want to do this, it can use the *intermediate CA* to sign _another_ CA, which will be exported as `nifi-cert.pem` and `nifi-key.key`. The only requirement here is that a certificate which the company trusts and has the capability to sign other certificates is exported to this format. 
* The *intermediate CA* as composed of `nifi-cert.pem` and `nifi-key.key` will be inserted into all truststores generated by the TLS toolkit. Because this is the certificate which explicitly signed the *node* certificate, any certificate chain validation will end here and return successfully. 
* The *root CA* does not need to be present in any generated truststore. 
* The Java default truststore (`$JAVA_HOME/jre/lib/security/cacerts`) does not enter into this process at all.

Reply:
I was thinking about how we can rollout a new CA intermediate before the old one expire.

If we add the signing external CA (referred to as the root, but I beleive it can be an intermediate as well) to the truststore we can replace the CA gradually without a complete downtime.

Is that correct @alopresto ?

Reply:
Yes, if the *root CA* signs *old/current intermediate CA* and *new CA*, and *root CA* is in the NiFi truststore, then certificates signed by both *old intermediate CA* and *new CA* will be trusted. You can read more about CA rollover in [IETF RFC 6489](https://tools.ietf.org/html/rfc6489). 

Currently we do not and have no plans to insert the *root CA* certificate into the truststore. This will require new feature work if it is desired.